### ISSUE TITLE
Zeros in the prediction

### ISSUE THREAD

**Issue by guptavasu1213** [msminhas93/DeepLabv3FineTuning#6]
In the variable `data`, I get all zeros for some reason. Can you please provide assistance regarding that?

```
import torch
model = torch.load('output/weights.pt')
model.eval()

import urllib
url, filename = ("file:///home/rhobincu/gitroot/DeepLabv3FineTuning/CrackForest/Images/092.jpg", "092.jpg")
try: urllib.URLopener().retrieve(url, filename)
except: urllib.request.urlretrieve(url, filename)

# sample execution (requires torchvision)
from PIL import Image
from torchvision import transforms
input_image = Image.open(filename)
print(input_image)
preprocess = transforms.Compose([
    [transforms.Resize((512,512)), transforms.ToTensor()]
)

input_tensor = preprocess(input_image)
input_batch = input_tensor.unsqueeze(0) # create a mini-batch as expected by the model

# move the input and model to GPU for speed if available
if torch.cuda.is_available():
    print('Using GPU!')
    input_batch = input_batch.to('cuda')
    model.to('cuda')

import time

start = time.clock()
with torch.no_grad():
    output = model(input_batch)['out'][0]
end = time.clock()
print('Inference duration (s): ', end - start)
data = output.argmax(0).detach().byte().cpu().numpy()

# create a color pallette, selecting a color for each class
palette = torch.tensor([2 ** 25 - 1, 2 ** 15 - 1, 2])
colors = torch.as_tensor([i for i in range(2)])[:, None] * palette
colors = (colors % 255).numpy().astype("uint8")
print(colors)

# plot the semantic segmentation predictions
img_size = input_image.size
r = Image.fromarray(data).resize(img_size)
r.putpalette(colors)

import matplotlib.pyplot as plt
plt.imshow(r)
plt.show()
```

**Comment by msminhas93** [comment]
I think `data = output.argmax(0).detach().byte().cpu().numpy()` this line is quantizing the output to zeros. You'll have to use a threshold. I don't understand the rationale behind using argmax. Also if you are not resizing during training don't resize during inference.

```
import torch, os
model = torch.load(r".\CFExp\weights.pt")
model.eval()
model.to('cuda')

# sample execution (requires torchvision)
from PIL import Image
from torchvision import transforms
input_image = Image.open(r".\CrackForest\Images\092.jpg")
print(input_image)
preprocess = transforms.Compose([transforms.ToTensor()])

input_tensor = preprocess(input_image)
input_batch = input_tensor.unsqueeze(0) # create a mini-batch as expected by the model
print(input_batch.shape)
# move the input and model to GPU for speed if available
if torch.cuda.is_available():
    print('Using GPU!')
    input_batch = input_batch.to('cuda')
    model.to('cuda')

import time

start = time.clock()
with torch.no_grad():
    output = model(input_batch)['out']
end = time.clock()
print('Inference duration (s): ', end - start)
data = output.detach().cpu().numpy()

# create a color pallette, selecting a color for each class
r = Image.fromarray(data[0][0]>0.1)

plt.imshow(r)
```

![image](https://user-images.githubusercontent.com/6465972/87202931-8e525280-c2cf-11ea-8a5f-cc6038062288.png)
